### PR TITLE
Use the document base url when resolving iframe URLs

### LINKED ISSUE #10576: Use the document base url when resolving iframe URLs
According to [the specification](https://html.spec.whatwg.org/multipage/embedded-content.html#otherwise-steps-for-iframe-or-frame-elements) (step 1), we need to be [using the base URL](https://html.spec.whatwg.org/multipage/infrastructure.html#parse-a-url). This means calling `document.base_url()` instead of `window.get_url()` as the code currently does.

Code: `components/script/dom/htmliframeelement.rs`
Test:
```html
<base href="resources/"></base>
<iframe href="page.html">
```
Visit this from a web server (eg. `python -m SimpleHTTPServer`) with a `resources/` directory that contains a `page.html` file that is not empty.

There does not appear to be an automated test for this yet, so we'll need to [write a reference test](https://github.com/servo/servo/blob/master/tests/wpt/README.md#writing-new-tests)! It belongs in `tests/wpt/web-platform-tests/html/semantics/embedded-content/the-iframe-element/`.

### PR DESCRIPTION
**Pull request by ziyunli**
Fixes #10576.

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10626)

### CODE REVIEW

File: components/script/dom/htmliframeelement.rs [hunk 1]
```diff
@@ -82,8 +82,8 @@ impl HTMLIFrameElement {
             if url.is_empty() {
                 None
             } else {
-                let window = window_from_node(self);
-                window.get_url().join(&url).ok()
+                let document = document_from_node(self);
+                document.base_url().join(&url).ok()
```
Comment (KiChjang) [inline components/script/dom/htmliframeelement.rs]: Looks like no other code is referencing `document` at all, can you make this a one-liner instead?

Comment (sliz1) [inline components/script/dom/htmliframeelement.rs]: sounds good, I will update this PR

**Comment by KiChjang** [comment]
@bors-servo r+

Thanks!

**Comment by KiChjang** [comment]
@bors-servo retry
* #10549

**Pull request merged**